packages/istanbul-report-modern-html/src/summaryTableLine.js

```diff
@@ -0,0 +1,188 @@
+import * as React from 'react';
+
+function ShowPicture({ num }) {
+    let rest;
+    let cls = '';
+    if (isFinite(num)) {
+        if (num === 100) {
+            cls = ' cover-full';
+        }
+        num = Math.floor(num);
+        rest = 100 - num;
+        return (
+            <>
+                <div
+                    className={'cover-fill' + cls}
+                    style={{ width: num + '%' }}
+                />
+                <div className="cover-empty" style={{ width: rest + '%' }} />
+            </>
+        );
+    } else {
+        return false;
+    }
+}
+
+function MetricCells({ metrics }) {
+    return (
+        <>
+            <td className={'pct ' + metrics.classForPercent}>{metrics.pct}%</td>
+            <td className={'abs ' + metrics.classForPercent}>
+                {metrics.covered}
+            </td>
+            <td className={'abs ' + metrics.classForPercent}>
+                {metrics.total}
+            </td>
+        </>
+    );
+}
+
+function FileCell({
+    file,
+    prefix,
+    expandedLines,
+    setExpandedLines,
+    hasChildren,
+    setFileFilter
+}) {
+    if (hasChildren) {
+        const expandedIndex = expandedLines.indexOf(prefix + file);
+        const isExpanded = expandedIndex >= 0;
+        const newExpandedLines = isExpanded
+            ? [
+                  ...expandedLines.slice(0, expandedIndex),
+                  ...expandedLines.slice(expandedIndex + 1)
+              ]
+            : [...expandedLines, prefix + file];
+
+        return (
+            <>
+                <a
+                    href="javascript:void(0)"
```

Comment:
Hm... Is this necessary? I would rather use an unstyled (css-reset) button to indicate actions. ideally `<a>`s are for navigation right?

Reply:
I changed the expand button to button. The other link I've kept `<a>` for consistency with the file links (and it does change the page and url anyway)